### PR TITLE
#181414755 Ft user registration

### PR DESCRIPTION
**Pull request by itisWasp**
# #181414755 On successful registration, users should receive a JWT

### What does this PR do?
Register a user and generate a jwt token when an account is successful created.
### Description of Task to be completed?
Once a Users is registered via the registration endpoint, receives a JWT to be required on all subsequent calls to all other endpoints that require authentication.

The User receives a descriptive error message when they enter invalid parameters in an attempt to signup.
if a user has an 
existing user email,
invalid email or password, he is prompted with the appropriate error message. on the other hand the 
password of length less than 8, and a password must contain an Upper case, a Number, a special character to be recognised as strong.

### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - switch to develop branch
  - ``yarn install``: to install all dependencies
  - ``yarn dev``: to start the development server
  -   Open postman
  - Navigate to ``http://localhost:3000/api/register``: use ``POST`` http verb  to input the data from the body to be sent to the server example:``{
    "FirstName": "Jane",
    "LastName": "Doe",
    "Email": "test@gmail.com",
    "UserName" : "Doe",
    "Password" : "Password@2022"
}``
  - Create .env file
  - add database url using "DB_CONNECT" as variable
- Navigate to heroku page of the team and open Heroku Postgress DB go to the settings and check the URI of the DB paste it in  "DB_CONNECT" variable.

### What are the relevant pivotal tracker stories?
[#181414755](https://www.pivotaltracker.com/story/show/181414755)

### CODE REVIEW

**Comment by ericus123** [comment]
Rebase from develop

**Comment by ericus123** [comment]
The indentation doesn't match the one configured in prettier.
Are you using prettier as your code formatter? if not, make sure you use it.

**Review by didasmbalanya — CHANGES_REQUESTED** [review]
Please try and resolve the comments from Hound CI.

File: src/controllers/userController.js [hunk 1]
```diff
@@ -1,11 +1,44 @@
-import models from '../database/models';
-const User = models.User;
+import models from '../database/models'
+import createUserService from '../services/UserServices'
+import bcrypt from 'bcrypt'
```
[the remark this answers is not in the document]
Comment (Robertishimwe) [reply src/controllers/userController.js]: Add semicolon

**Pull request closed without merging**